code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm

```diff
@@ -376,11 +432,32 @@
 		M.bodytemperature = min(310, M.bodytemperature - (adj_temp * TEMPERATURE_DAMAGE_COEFFICIENT))
 
 // Juices
+/datum/reagent/drink/juice/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
+	..()
+
+	var/effective_dose = dose/2
+	if(issmall(M))
+		effective_dose *= 2
+
+	if(alien == IS_UNATHI)
+		if(effective_dose < 2)
```

Comment:
This stuff was cleaned up in a recent Bay PR, it probably wouldn't hurt to grab that.